### PR TITLE
Fix BList::append_lazy, and dogfood proto::DList

### PR DESCRIPTION
**Pull request by Gankro**
I want to use proto::DList's cursors in BList, but was running into lexical-scope issues with the algorithms. For now I'm just doing the migration and calling it a day.

### CODE REVIEW

File: src/blist.rs [hunk 1]
```diff
@@ -722,6 +728,36 @@ mod test {
...
+        let mut v = list_from(&[10, 20, 30, 40, 50, 60, 70, 80, 90, 100, 110]);
+        let w = list_from(&[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11,
+                            10, 20, 30, 40, 50, 60, 70, 80, 90, 100, 110]);
+        let x = list_from(&[]);
+
+        // Normal append
+        u.append_lazy(&mut v);
+        assert_eq!(u.len(), 22);
+        assert_eq!(v.len(), 0);
+        assert_eq!(&u, &w);
```
Comment (apasel422) [inline src/blist.rs]: Do all these `assert_eq`s need to use references?

Comment (Gankro) [inline src/blist.rs]: In my head I was just like "anything else will move the lists out", but in retrospect the macro does auto-reffing, I guess?

Comment (apasel422) [inline src/blist.rs]: Yeah, autoref should work.

Comment (Gankro) [inline src/blist.rs]: Fixed (although cargo's busted on my computer right now :( )

**Comment by apasel422** [comment]
@Gankro Looks good to me.

**Comment by Gankro** [comment]
@reem my cargo is busted, so I added you as an owner (only temporarily, if you don't want to deal with that longterm) if you want to do a new release.

**Pull request merged**